app/models/manageiq/providers/base_manager/event_catcher.rb

```diff
@@ -15,4 +15,8 @@ def friendly_name
       end
     end
   end
+
+  def self.kill_priority
+    130
```

Comment:
Wonder if we should encapsulate these numbers into a set of constants, like HIGH MEDIUM LOW.  We did something similar with queue priorities: https://github.com/ManageIQ/manageiq/blob/1e9398f43e778bf9ded7209338b06926d3d64cf3/app/models/miq_queue.rb#L26-L30

I guess it depends on the next iteration of this code, cause I think you were thinking a YAML file or something more declarative.

Reply:
Yeah, this is an area where I wanted to get some input on what exactly was being accomplished with the ordering and if we could simplify the order.

Right now I grouped the workers pretty conservatively to preserve the previous order, but it wasn't clear which of the non-typed workers could be grouped (for example does the UI worker need to be before the remote console worker, or could those be a group?).

Reply:
And I'm not sure about the file bit for this now that I'm writing it out. I can't remember what the justification was for that direction.

Reply:
As discussed offline, but I'll reiterate here...

The purpose of the kill order was to react to critical level of resources (usually low memory) on the system.  The idea was that when in a critical system situation, we would kill "less important" or "high memory" workers first, trying to balance between those 2, also taking into account that certain workers are also on other appliances, so killing on one appliance might be ok, cause the work can be picked up elsewhere.

It's really less about specific worker classes, and more about the features/roles of those workers.  This list would be better if instead of...

```
MIQ_WORKER_TYPES_IN_KILL_ORDER = %w(
  MiqEmsMetricsProcessorWorker
  ManageIQ::Providers::Azure::CloudManager::MetricsCollectorWorker
  ManageIQ::Providers::Amazon::CloudManager::MetricsCollectorWorker
  ManageIQ::Providers::Redhat::InfraManager::MetricsCollectorWorker
  ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCollectorWorker
  ManageIQ::Providers::Openshift::ContainerManager::MetricsCollectorWorker
  ManageIQ::Providers::Google::CloudManager::MetricsCollectorWorker
  ManageIQ::Providers::Vmware::InfraManager::MetricsCollectorWorker
  ManageIQ::Providers::Openstack::CloudManager::MetricsCollectorWorker
  ManageIQ::Providers::Openstack::NetworkManager::MetricsCollectorWorker
  ManageIQ::Providers::Redhat::NetworkManager::MetricsCollectorWorker
  ManageIQ::Providers::Openstack::InfraManager::MetricsCollectorWorker
  MiqReportingWorker
  MiqSmartProxyWorker
  MiqGenericWorker
  MiqEventHandler
  ...
```

it were

```
MIQ_WORKER_TYPES_IN_KILL_ORDER = %w(
  metrics_processors
  metrics_collectors
  reporting
  smart_proxies
  generic
  event_handlers
  ...
```